### ISSUE TITLE
Use MultiHeadAttention in tf:2.3.1

### ISSUE THREAD

**Issue by yananchen1989** [tensorflow/tensorflow#48936]
I want to use `MultiHeadAttention` layer in tf:2.3.1 due to CUDA version limit.

here is the test code:
```
import multi_head_attention  

test_layer = multi_head_attention.MultiHeadAttention(
    num_heads=12, key_dim=64)
# Create a 3-dimensional input (the first dimension is implicit).
query = keras.Input(shape=(40, 80))
output, coef = test_layer(query, query, return_attention_scores=True)

```

I copy this file : https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/python/keras/layers/multi_head_attention.py 
to my current working path for import.

Here is the error:

> ---------------------------------------------------------------------------
> TypeError                                 Traceback (most recent call last)
> <ipython-input-6-8ef8346e0a54> in <module>
>       1 query = keras.Input(shape=(40, 80))
> ----> 2 output, coef = test_layer(query, query, return_attention_scores=True)
> 
> ~/yanan/env/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py in __call__(self, *args, **kwargs)
>     924     if _in_functional_construction_mode(self, inputs, args, kwargs, input_list):
>     925       return self._functional_construction_call(inputs, args, kwargs,
> --> 926                                                 input_list)
>     927
>     928     # Maintains info about the `Layer.call` stack.
> 
> ~/yanan/env/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py in _functional_construction_call(self, inputs, args, kwargs, input_list)
>    1115           try:
>    1116             with ops.enable_auto_cast_variables(self._compute_dtype_object):
> -> 1117               outputs = call_fn(cast_inputs, *args, **kwargs)
>    1118
>    1119           except errors.OperatorNotAllowedInGraphError as e:
> 
> ~/yanan/env/lib/python3.6/site-packages/tensorflow/python/autograph/impl/api.py in wrapper(*args, **kwargs)
>     256       except Exception as e:  # pylint:disable=broad-except
>     257         if hasattr(e, 'ag_error_metadata'):
> --> 258           raise e.ag_error_metadata.to_exception(e)
>     259         else:
>     260           raise
> 
> TypeError: in user code:
> 
>     /root/yanan/berts/topic_classification_augmentation/multi_head_attention.py:510 call  *
>         attention_output, attention_scores = self._compute_attention(
>     /root/yanan/berts/topic_classification_augmentation/multi_head_attention.py:475 _compute_attention  *
>         attention_scores = self._masked_softmax(attention_scores, attention_mask)
>     /root/yanan/berts/topic_classification_augmentation/multi_head_attention.py:438 _masked_softmax  *
>         return self._softmax(attention_scores, attention_mask)
>     /root/yanan/env/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py:926 __call__  **
>         input_list)
>     /root/yanan/env/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py:1117 _functional_construction_call
>         outputs = call_fn(cast_inputs, *args, **kwargs)
> 
>     TypeError: call() takes 2 positional arguments but 3 were given

Any solution ?

**Comment by yananchen1989** [comment]
In tf:2.4.1 everything is OK:

`att = layers.MultiHeadAttention(num_heads=num_heads, key_dim=embed_dim)`

**Comment by saikumarchalla** [comment]
@yananchen1989 The file you are using referring to 2.4 version but the multi_head_attention.py  doesn't exist in TF 2.3.1.Thanks!

**Comment by yananchen1989** [comment]
> @yananchen1989 The file you are using referring to 2.4 version but the multi_head_attention.py doesn't exist in TF 2.3.1.Thanks!

I copy this file to my working path manually, aiming to import it from current path by 
`import multi_head_attention `

**Comment by saikumarchalla** [comment]
@yananchen1989  Seems the multi head attention layer implemented from TF 2.4 version and that's why you are encountering the issues in 2.3. Thanks!

**Comment by saikumarchalla** [comment]
@yananchen1989 Closing the issue since it was resolved. Please feel free to re-open  the issue if you have any further queries.
